internal/pkg/describe/env.go

```diff
@@ -50,23 +64,70 @@ func NewEnvDescriber(projectName string, envName string) (*EnvDescriber, error)
 		return nil, fmt.Errorf("assuming role for environment %s: %w", env.ManagerRoleARN, err)
 	}
 	return &EnvDescriber{
-		env:          env,
-		store:        svc,
-		apps:         apps,
-		sessProvider: sess,
+		env:            env,
+		store:          svc,
+		proj:           proj,
+		apps:           apps,
+		rgClient:       resourcegroups.New(sess),
+		cloudformation: cloudformation.New(sess),
 	}, nil
 }
 
 // Describe returns info about a project's environment.
 func (e *EnvDescriber) Describe() (*EnvDescription, error) {
-	var tags map[string]string
+	appsForEnv, err := e.FilterAppsForEnv()
+	if err != nil {
+		return nil, err
+	}
+
 	return &EnvDescription{
 		Environment:  e.env,
-		Applications: e.apps,
-		Tags:         tags,
+		Applications: appsForEnv,
+		Tags:         e.proj.Tags,
 	}, nil
 }
 
+func (e *EnvDescriber) FilterAppsForEnv() ([]*archer.Application, error) {
+	var appObjects []*archer.Application
+
+	tags := map[string]string{
+		"ecs-environment": e.env.Name,
+	}
+	arns, err := e.rgClient.GetResourcesByTags(cloudformationResourceType, tags)
+	if err != nil {
+		return nil, err
+	}
+
+	var stacksOfEnvironment []string
```

Comment:
What do you think maybe like this?

``` go
...
        stacksofEnvironment := make(map[string]bool)
	for _, arn := range arns {
		stack, err := e.parseARN(arn)
		if err != nil {
			return nil, err
		}
		stacksOfEnvironment[stack] = true
	}
        
	for _, app := range e.apps {
		stackName := stack.NameForApp(e.proj.Name, e.env.Name, app.Name)
		if  stacksOfEnvironment[stackName] {
				appObjects = append(appObjects, app)
                }
	}
...
```

Reply:
Ah, I see. That avoids the nested for loops. Much better! Thanks!